runtime/common/src/lib.rs

```diff
@@ -80,6 +94,25 @@ impl WeightToFeePolynomial for WeightToFee {
 	}
 }
 
+/// Maps the proof size component of `Weight` to a fee.
+pub struct ProofSizeToFee;
+impl WeightToFeePolynomial for ProofSizeToFee {
+	type Balance = Balance;
+
+	fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
+		// Map 1MB proof to 1 UNIT.
+		let p = UNIT;
+		let q = 10_000_000;
```

Comment:
1MB should be `1_000_000` instead of `10_000_000`. https://www.gbmb.org/megabytes